tests/hassh Add test cases for hassh

MalakhatkoVadym opened the pull request:
This commit adds test cases for the ssh.hassh, ssh.hassh.string, ssh.hassh.server, ssh.hassh.server.string keywords.

Modifies https://github.com/OISF/suricata-verify/pull/263 by
1. Fixing required min-version for tests.

victorjulien commented:
In the `ssh-hassh` test I see that the client and server hashes are exactly the same:
```json
{
  "client": {
    "proto_version": "2.0",
    "software_version": "OpenSSH_for_Windows_7.7",
    "hassh": "2dd6531c7e89d3c925db9214711be76a",
    "hassh.string": "curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,ext-info-c;chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com;umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1;none"
  },
  "server": {
    "proto_version": "2.0",
    "software_version": "OpenSSH_7.4",
    "hassh": "2dd6531c7e89d3c925db9214711be76a",
    "hassh.string": "curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,ext-info-c;chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com;umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1;none"
  }
}
```
Is that correct? The software versions are different versions of OpenSSH.

MalakhatkoVadym commented:
> In the `ssh-hassh` test I see that the client and server hashes are exactly the same:
> 
> ```json
> {
>   "client": {
>     "proto_version": "2.0",
>     "software_version": "OpenSSH_for_Windows_7.7",
>     "hassh": "2dd6531c7e89d3c925db9214711be76a",
>     "hassh.string": "curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,ext-info-c;chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com;umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1;none"
>   },
>   "server": {
>     "proto_version": "2.0",
>     "software_version": "OpenSSH_7.4",
>     "hassh": "2dd6531c7e89d3c925db9214711be76a",
>     "hassh.string": "curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,ext-info-c;chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com;umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1;none"
>   }
> }
> ```
> 
> Is that correct? The software versions are different versions of OpenSSH.

No, my bad, there was a bug in rust ssh logger. Fixed and updated hassh pr. Thanks.  
https://github.com/OISF/suricata/pull/5141

MalakhatkoVadym commented:
https://github.com/OISF/suricata-verify/pull/267
